packages/i18n/src/index.js

```diff
@@ -34,7 +35,9 @@ export function setLocaleData( localeData = { '': {} }, domain = 'default' ) {
 		} );
 	}
 
-	i18n.options.locale_data[ domain ] = localeData;
+	i18n.options.locale_data[ domain ] = has( i18n.options.locale_data, domain )
+		? merge( {}, i18n.options.locale_data[ domain ], localeData )
+		: localeData;
```

Comment:
Seems reasonable to me, A second opinion might be good. I wonder if we do need the deep `merge` or if it's possible to use just some `Object.assign` or similar. I don't have enough knowledge about the locale data format to answer personally.

Also, should we add a unit test?

Reply:
Got thinking about this, the other concern we might have here is protecting the `[""]` index in the `localeData`.  We probably don't want plugins to trounce the `locale_data[""][domain]` property right.  I'm not sure of what approach should be taken here yet.

Reply:
It's probably good argument for making sure Jed is ALWAYS instantiated with a `domain` (even if its extracted from `locale_data['']['domain']` if it exists and thus `i18n.options.locale_data[domain]` will remain consistent with `locale_data['']['domain']`?

Reply:
>We probably don't want plugins to trounce the `locale_data[""]['domain']` property right?

What consequences would there be, out of curiosity?

I guess in my head I'm wondering, if a plugin developer erroneously set locale data for the default domain, with the merging behavior would it just be as though they're adding the existing set?

I'd be fine to be explicit with passing the argument, though.

Reply:
>...it looks like 'some_other_domain' is never really referenced anywhere by jed (its just ignored).

Interesting. I guess it makes sense, since we're already kinda doing it wrong in Gutenberg:

```html
<script type='text/javascript'>
wp.i18n.setLocaleData( {"":{"domain":"gutenberg","lang":"en_US"}} );
</script>
```

It's assigning data for the `default` locale (by omitting the second argument), but its own `localeData[ '' ].domain` is set to `gutenberg`. Seems it should be `default`, yeah?

>However there is the possibility of a plugin author messing up the plural_forms or lang properties if they are interacting with setLocale directly and not going through an api we add to core. So I'm wondering if we should just add some protection for that, or should we bother?

I'm not entirely clear what you're referring to by "setLocale directly". In any case, I wouldn't overthink it too much, unless it's a [footgun](https://en.wiktionary.org/wiki/footgun).

---

In any case, the remaining question I have here is if we should do as Riad originally suggested, and use `Object.assign` to do the merge? Pulling in Lodash's `merge` could be costly, when we don't need the deep merge.

Reply:
> Seems it should be default, yeah?

Yeah I agree.  But since its ignored we probably don't even need to set `[''].domain`.

> I'm not entirely clear what you're referring to by "setLocale directly"

Well the statement I made is on the assumption that something will land in WordPress core that will automate queuing up `wp.i18n.setLocale()` via `wp_inline_script` or some other mechanism.  But that wouldn't rule out some plugin/code calling `wp.i18n.setLocale` directly within its own js (as opposed to indirectly through whatever api wp core exposes.

> In any case, the remaining question I have here is if we should do as Riad originally suggested, and use Object.assign to do the merge? Pulling in Lodash's merge could be costly, when we don't need the deep merge.

Agreed, and I forgot about Riad's comment (thanks for the reminder).  Pushed the change ⚡️